### PR TITLE
fix #472 - make tables sharable with organizations

### LINKED ISSUE #472: Share Table with Org
### Story
User wants to share their Table Org wide

### Acceptance Criteria
GIVEN a user has a table they want to share
WHEN they go to the Table/View page
THEN they are given the option to share with their Org

### PR DESCRIPTION
**Pull request by yilmazbekdemir**
## Purpose
Users can share their tables with their organizations

## Approach
Added to share_with_organization checkbox to silo edit page.

_Related ticket: #472

### CODE REVIEW

File: silo/views.py [hunk 1]
```diff
@@ -430,7 +430,7 @@ def appendTwoSilos(mapping_data, lsid, rsid, msid):
 # Edit existing silo meta data
 @csrf_protect
 @login_required
-def editSilo(request, id):
+def edit_silo(request, id):
```
[the remark this answers is not in the document]
Comment (yilmazbekdemir) [reply silo/views.py]: I changed name according to PEP8. https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

File: silo/migrations/0043_auto_20180503_0625.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    operations = [
+        migrations.AddField(
+            model_name='silo',
+            name='share_with_organization',
+            field=models.BooleanField(default=False),
+        ),
+        migrations.AddField(
+            model_name='silo',
+            name='share_with_workflow_teams',
```
[the remark this answers is not in the document]
Comment (yilmazbekdemir) [reply silo/migrations/0043_auto_20180503_0625.py]: Yes, you are right. I updated to model with also share_with_workflow_teams field but realized that this is for other ticket but I forgot the clean migration script.

File: silo/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jefmoura) [reply silo/views.py]: why did you apply PEP8 on it if you didn't change anything in the function?

File: silo/forms.py [hunk 3]
```diff
@@ -54,8 +54,8 @@ def __init__(self, user=None, *args, **kwargs):
 
     class Meta:
         model = Silo
-        fields = ['id', 'name', 'description', 'tags', 'shared', 'owner',
-                  'workflowlevel1']
+        fields = ['id', 'name', 'description', 'tags', 'shared',
+                  'share_with_organization','owner', 'workflowlevel1']
```
Comment (jefmoura) [inline silo/forms.py]: It's missing a space between the comma and the single quote.

File: silo/tests/test_views.py [hunk 4]
```diff
@@ -1082,7 +1083,199 @@ def test_private_silo_detail_with_owner_user(self):
...
+
+        url = reverse('list_silos')
+
+        request = self.factory.get(url)
+        request.user = self.user
+        response = views.list_silos(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, 'Test Share Silo')
+
+    def test_list_public_silo_with_unshared_user(self):
```
Comment (jefmoura) [inline silo/tests/test_views.py]: The name should be just `test_list_public_silo`, don't you think?

**Pull request merged**